h/static/scripts/tests/controllers/search-bar-controller-test.js

```diff
@@ -172,4 +180,38 @@ describe('SearchBarController', function () {
     input.dispatchEvent(new Event('input'));
     assert.isFalse(dropdown.classList.contains('is-open'));
   });
+  /*
+  it('creates a lozenge for valid input "foo"', function () {
+    var testStr = 'foo:';
+    var e1 = new Event('keydown');
+    e1.keyCode = 32;
+    input.dispatchEvent(new Event('focusin'));
+    input.value = testStr;
+    input.dispatchEvent(new Event('input'));
+    input.dispatchEvent(e1);
+    assert.equal(testEl.querySelector('.lozenge-content').textContent, testStr);
+  });
+*/
+
+  unroll('should create lozenge on space', function (fixture) {
+    var e1 = new Event('keydown');
```

Comment:
I would suggest that we take advantage of the `syn` library introduced in https://github.com/hypothesis/h/pull/3956 to make sure that input is done in a more realistic way, and with less code.